on-prem-docs/migration.md

```diff
@@ -0,0 +1,251 @@
+# Migrate to a Deployment of Automate Builder
+
+Long Term Support for On-Prem Builder is provided for installations of [Builder via Chef Automate](https://automate.chef.io/docs/on-prem-builder/).
+This document will guide you through the steps necessary to migrate your On-Prem Builder data into an Automate Builder.
+
+NOTE: The migration will overwrite any existing data in Automate Builder (target) with the data from your current On-Prem (source) installation. The assumption is that the Automate Builder target node is not yet in use.
+
+## Preparing for a Migration
+
+The data that Builder stores is fairly lightweight and thus the migration strategy is pretty straightforward. On-Prem Builder has two types of data that will need to be migrated:
+
+1. Minio/S3 habitat artifacts
+1. PostgreSQL package and user metadata
+
+The minio/S3 data will be copied to the Automate Builder target via minio mirroring utility.
+
+The migration of the PostgreSQL data will be done via a `pg_dump` on the source and then restoring with `psql` on the target.
+
+IMPORTANT: For this migration, you will need to configure Chef Automate as an [OAuth Provider for Habitat Builder](https://automate.chef.io/docs/applications-setup/#authenticating-existing-chef-automate-and-builder-installations). Following the PostgreSQL data migration, you will need to create local accounts in Chef Automate that match the userids from your source Builder instance.
+
+### Creating a fallback copy of Chef Automate Builder data
+
+Since the data migration is destructive and will overwrite any previous Builder data on the target, perform a backup in case the original state needs to be restored:
+
+```
+sudo chef-automate backup create
+```
+
+### Validating Versions
+
+It is the Builder API that runs database migrations and is responsible for making schema changes, ensuring that the PostgreSQL tables are all up to date.
+Check that your target Automate Builder instance is running the same or newer Builder API version than your current On-Prem Builder (source). This is required to ensure that there are no PostgreSQL schema incompatibilities. The Builder API service on the target Automate Builder node will run any migrations necessary to update the PostgreSQL data and schemas to the correct format. Therefore the Automate Builder target must be the same or newer version.
+
+To check the API version installed on the source and target Builder nodes run:
+
+```
+hab pkg path habitat/builder-api
```

Comment:
Nit:  This will show the latest installed version which _may_ be different than what is running.  Maybe consider `hab svc status`?

Reply:
Great suggestion. Addressed via https://github.com/habitat-sh/on-prem-builder/pull/247/commits/161b40423ce5dfe30e11905fef68e37bc6ef894b